sql/analyzer/triggers.go

```diff
@@ -273,6 +276,20 @@ func applyTrigger(ctx *sql.Context, a *Analyzer, originalNode, n sql.Node, scope
 				}), transform.NewTree, nil
 			}
 		case *plan.DeleteFrom:
+			// TODO: This should work correctly when there is only one table that
+			//       has a trigger on it, but it won't work if a DELETE FROM JOIN
+			//       is deleting from two tables that both have triggers. Seems
+			//       like we need something like a MultipleTriggerExecutor node
+			//       that could execute multiple triggers on the same row from its
+			//       wrapped iterator. There is also an issue with running triggers
+			//       because their field indexes assume the row they evalute will
+			//       only ever contain the columns from the single table the trigger
+			//       is based on, but this isn't true with UPDATE JOIN or DELETE JOIN.
+			if n.HasExplicitTargets() {
```

Comment:
the prepend/strip scope pattern comes to mind, but it would be nice to not do that

Reply:
I briefly thought about prepend/strip, but I don't think it would work here... we could have multiple triggers from multiple tables that need to execute on the same row, so it seems messy to try and strip the row apart into the right table-specific components for each trigger, then reassemble it all on the way out.